### PR TITLE
Ms.wlt node unsynced

### PR DESCRIPTION
**Pull request by mariano54**
Fixes the issues with the wallet not syncing in untrusted mode, if connected to a trusted peer that is not synced.

Also some other important sycing fixes. This should be merged along with: https://github.com/Chia-Network/chia-blockchain/pull/12433

This needs to go into 1.5.1 as well

### CODE REVIEW

File: chia/wallet/util/wallet_sync_utils.py [hunk 1]
```diff
@@ -324,27 +324,29 @@ async def request_header_blocks(
     peer: WSChiaConnection, start_height: uint32, end_height: uint32
 ) -> Optional[List[HeaderBlock]]:
     if Capability.BLOCK_HEADERS in peer.peer_capabilities:
+        log.warning(f"Peer {peer.get_peer_info()} has the capability! {peer.peer_capabilities}")
```
Comment (trepca) [inline chia/wallet/util/wallet_sync_utils.py:327]: log.warning? is this just to test?

**Review by trepca — DISMISSED** [review]
LGTM, just `log.warning` seems a bit excessive for what seems to be more debug like messages

File: chia/wallet/util/wallet_sync_utils.py (hunk 1 above)
Comment (mariano54) [reply chia/wallet/util/wallet_sync_utils.py:327]: Oh this was debugging, I'll revert

**Review by trepca — DISMISSED** [review]

File: chia/wallet/util/wallet_sync_utils.py (hunk 1 above)
Comment (wallentx) [reply chia/wallet/util/wallet_sync_utils.py]: Resolving conversation, as this looks to be fixed.

**Comment by mariano54** [comment]
Force pushed to get rid of unsigned commits

**Review by trepca — APPROVED** [review]

**Review by Quexington — APPROVED** [review]

**Pull request merged**